### PR TITLE
Introduce the KCC provider for servicemapping and sample generation in Config Connector

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
Add a Terraform KCC provider to generate KCC servicemappings and TF samples automatically.

Used the Terraform provider to reuse the existing logic that reads and parses the api.yaml files and the additional terraform.yaml layer.

Context: b/258035442

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by maqiuyujoyce** [comment]
I didn't check any checkbox as this change is not for the Terraform provider. Let me know if there is anything else I should do about this PR.

File: mmv1/provider/terraform_kcc.rb [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# Copyright 2019 Google Inc.
```
Comment (rileykarson) [inline mmv1/provider/terraform_kcc.rb:1]:
```suggestion
# Copyright 2023 Google Inc.
```

File: mmv1/provider/terraform_kcc.rb [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          self
+        )
+      end
+    end
+
+    def generate_resource_tests(pwd, data) end
+
+    def generate_resource_sweepers(pwd, data) end
+
+    def generate_iam_policy(pwd, data, generate_code, generate_docs)end
```
Comment (rileykarson) [inline mmv1/provider/terraform_kcc.rb:57]:
```suggestion
    def generate_iam_policy(pwd, data, generate_code, generate_docs) end
```

**Review by rileykarson — APPROVED** [review]
Looks reasonable to me! I'm not too familiar with the modern ServiceMapping format so I can't validate that very well, but the parts I am familiar with make sense. Two minor nitpicks (a copyright and spacing), let me know if you'd rather I just merge this rather than wait on them getting updated.

File: mmv1/provider/terraform_kcc.rb (hunk 1 above)
Comment (maqiuyujoyce) [reply mmv1/provider/terraform_kcc.rb]: Good catch! Done.

File: mmv1/provider/terraform_kcc.rb (hunk 2 above)
Comment (maqiuyujoyce) [reply mmv1/provider/terraform_kcc.rb]: Done.

**Comment by maqiuyujoyce** [comment]
Thank you, @rileykarson ! Addressed the comments and updated the template to use the current year for copyright (instead of using a static year).

> I'm not too familiar with the modern ServiceMapping format so I can't validate that very well, but the parts I am familiar with make sense.

Thanks! We'll probably want to rely on the tests in config connector to do the validation.

File: mmv1/templates/kcc/product/service_mapping.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,132 @@
+<%
+autogen_exception
+-%>
+# Copyright <%= Time.new.year %> Google LLC
```
Comment (rileykarson) [inline mmv1/templates/kcc/product/service_mapping.yaml.erb:4]:
```suggestion
# Copyright 2023 Google LLC
```

File: mmv1/templates/kcc/samples/sample.tf.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright <%= Time.new.year %> Google LLC
```
Comment (rileykarson) [inline mmv1/templates/kcc/samples/sample.tf.erb:2]:
```suggestion
 * Copyright 2023 Google LLC
```

**Review by rileykarson — COMMENTED** [review]
Ah, some unit tests are complaining about the generated copyright dates. Let's just set static current ones, that should be fine.

We should be able to ignore the CLA check failure, since Google's the owner of your change anyways. I'll override.

**Review by rileykarson — APPROVED** [review]
Oh! The issue is a little different than I thought. We've got a hardcoded list of date formats in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/spec/copyright.rb, and `23` was only recently added as a suffix. In your branch (https://github.com/maqiuyujoyce/magic-modules/blob/master/mmv1/spec/copyright.rb) it's not included yet. While this fails right now, merging will resolve it.

LGTM

**Pull request merged**